File: app/src/main/java/knf/kuma/tv/anime/SyncPresenter.java
package knf.kuma.tv.anime;

import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;
import knf.kuma.tv.cards.SyncCardView;
import knf.kuma.tv.sync.SyncObject;

public class SyncPresenter extends Presenter {

    public SyncPresenter() {
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        return new ViewHolder(new SyncCardView(parent.getContext()));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, Object item) {
        ((SyncCardView) viewHolder.view).bind((SyncObject) item);
    }

    @Override
    public void onUnbindViewHolder(ViewHolder viewHolder) {

    }
}

File: app/src/main/java/knf/kuma/tv/anime/SyncPresenter.kt
package knf.kuma.tv.anime

import android.view.ViewGroup

import androidx.leanback.widget.Presenter
import knf.kuma.tv.cards.SyncCardView
import knf.kuma.tv.sync.SyncObject

class SyncPresenter : Presenter() {

    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        return ViewHolder(SyncCardView(parent.context))
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, item: Any) {
        (item as? SyncObject)?.let {
            (viewHolder.view as? SyncCardView)?.bind(it)
        }
    }

    override fun onUnbindViewHolder(viewHolder: ViewHolder) {

    }
}
